webtau-core/src/main/java/com/twosigma/webtau/data/table/Record.java

```diff
@@ -118,4 +147,27 @@ void add(Record record) {
             result.add(record);
         }
     }
+
+    private static class RecordFromStream {
+        private boolean hasMultiValues;
+        private boolean hasValueGenerators;
+        private List<Object> values;
+
+        public RecordFromStream(Stream<Object> valuesStream) {
+            values = new ArrayList<>();
+            Iterator<Object> it = valuesStream.iterator();
```

Comment:
Why not use `valueStream.forEach` instead of using the iterator manually?

Reply:
Will it require some manipulation with index wrapping that needs to be defined outside?

Reply:
OH nevermind, there is no more idx manipulation.